Sources/OktaOAuth2/Authentication/DeviceAuthorizationFlow.swift

```diff
@@ -319,14 +336,26 @@ extension DeviceAuthorizationFlow {
                     switch result {
                     case .failure(let error):
                         if case let APIClientError.serverError(serverError) = error,
-                           let oauthError = serverError as? OAuth2ServerError,
-                           oauthError.code == "authorization_pending"
+                           let oauthError = serverError as? OAuth2ServerError
                         {
-                            completion(.success(nil))
-                        } else {
-                            self.delegateCollection.invoke { $0.authentication(flow: self, received: .network(error: error)) }
-                            completion(.failure(error))
+                            switch oauthError.code {
+                            case "authorization_pending":
+                                // Keep polling, since we haven't received a token yet.
+                                completion(.success(nil))
+                                return
+                                
+                            case "slow_down":
+                                // Increase the polling interval for all subsequent requests, according to the specification.
+                                self.scheduleTimer(offsetBy: DeviceAuthorizationFlow.slowDownInterval)
```

Comment:
It seems like we need to capture  weak reference to `self` in the closure above. any thoughts ?

Reply:
The retain cycle is broken when the flow completes and `reset()` is called, but I see your point.